src/index.ts

```diff
@@ -280,14 +279,22 @@ function emitDom() {
 
   emitFlavor(webidl, new Set(knownTypes.Window), {
     name: "dom",
-    flavor: Flavor.Window,
-    global: "Window",
+    global: ["Window"],
     outputFolder,
   });
   emitFlavor(webidl, new Set(knownTypes.Worker), {
     name: "webworker",
-    flavor: Flavor.Worker,
-    global: "Worker",
+    global: ["Worker", "DedicatedWorker", "SharedWorker", "ServiceWorker"],
```

Comment:
Should we have a new `dedicatedworker` flavor that only contains `["DedicatedWorker", "Worker"]`? It would be nice if you're writing code for a dedicated worker to have the type checker disallow using service worker-only or shared worker-only APIs. But I can also understand that it may be difficult to teach users the difference between the "webworker" and "dedicatedworker" flavors.

Reply:
Sounds good to me and it was my initial plan. In the ideal world we could ultimately deprecate `webworker` in that case. I wonder what the maintainers think.